### PR TITLE
New genscript changes

### PR DESCRIPTION
**Pull request by mw-kapilg**
This PR contains the following new Genscript changes:

`useParallel` : Boolean

`strict`: Boolean

`loggingLevel`: String

- Default
- None
- Terse
- Concise
- Detailed
- Verbose 

`outputDetail`: String

- Default
- None
- Terse
- Concise
- Detailed
- Verbose

### CODE REVIEW

File: src/main/resources/templates/editMATLABTestTask.ftl [hunk 1]
```diff
@@ -25,6 +25,20 @@ extraUtility=addExecutableLink list=uiConfigSupport.getExecutableLabels('matlab'
...
+
+    [@ww.checkbox labelKey='matlab.test.use.parallel' name='useParallelChecked' toggle='true'/]
+
+    [@ww.select cssClass="builderSelectWidget" labelKey="matlab.test.output.detail" name="outputDetail"
+    list=["Default", "None", "Terse", "Concise", "Detailed", "Verbose"]/]
+
+    [@ww.select cssClass="builderSelectWidget" labelKey="matlab.test.logging.level" name="loggingLevel"
+    list=["Default", "None", "Terse", "Concise", "Detailed", "Verbose"]/]
+
+    [@ww.select cssClass="builderSelectWidget" labelKey="Sample" list=["a", "b"]/]
```
Comment (davidbuzinski) [inline src/main/resources/templates/editMATLABTestTask.ftl:39]: I think this got left in accidentally

File: src/main/resources/matlab-bamboo-plugin.properties [hunk 2]
```diff
@@ -33,6 +33,13 @@ matlab.test.coverage.exists= HTML code coverage
 matlab.model.coverage.exists= HTML model coverage
 matlab.code.coverage.directory= Folder path
 
+#Test run customizations
+matlab.test.customizations=Customize test run
+matlab.test.strict=Strict
+matlab.test.use.parallel=Use parallel
+matlab.test.output.detail=Output Detail
+matlab.test.logging.level=Logging level
```
Comment (davidbuzinski) [inline src/main/resources/matlab-bamboo-plugin.properties:41]: "Output detail" (lower case D) to be consistent with the other options

**Review by davidbuzinski — COMMENTED** [review]
Looks good! Just a couple minor changes

File: src/main/resources/matlab-bamboo-plugin.properties (hunk 2 above)
Comment (mw-kapilg) [reply src/main/resources/matlab-bamboo-plugin.properties:41]: Thanks for pointing it out, updated!

File: src/main/resources/templates/editMATLABTestTask.ftl (hunk 1 above)
Comment (mw-kapilg) [reply src/main/resources/templates/editMATLABTestTask.ftl:39]: Oops, removing it! Thanks

**Review by davidbuzinski — APPROVED** [review]
:shipit:

File: src/main/resources/matlab-bamboo-plugin.properties [hunk 3]
```diff
@@ -33,6 +33,13 @@ matlab.test.coverage.exists= HTML code coverage
 matlab.model.coverage.exists= HTML model coverage
 matlab.code.coverage.directory= Folder path
 
+#Test run customizations
+matlab.test.customizations=Customize test run
+matlab.test.strict=Strict
```
Comment (ppandian28) [inline src/main/resources/matlab-bamboo-plugin.properties:38]: @mw-kapilg do  check with @mw-hrastega if the UI strings are okay, if you haven't.

File: src/test/java/com/mathworks/ci/task/MatlabTestTaskTest.java [hunk 4]
```diff
@@ -101,14 +105,18 @@ public void testExectuteRunsWithProvidedArguments() throws TaskException, IOExce
...
         configurationMap.put("pdf", "report.pdf");
         configurationMap.put("html", "code-coverage");
         configurationMap.put("htmlModel", "model-coverage");
         configurationMap.put("stm", "results.mldatx");
         configurationMap.put("srcfolder", "src/:src1");
         configurationMap.put("testFolders", "test/");
-        configurationMap.put("testTag", "all");      
+        configurationMap.put("testTag", "all");
+        configurationMap.put("outputDetail", "Detailed");
+        configurationMap.put("loggingLevel", "Detailed");
```
Comment (ppandian28) [inline src/test/java/com/mathworks/ci/task/MatlabTestTaskTest.java:119]: Are we picking Detailed as the default level of verbosity? Is this consistent with Jenkins ? Do you know why?

**Review by ppandian28 — COMMENTED** [review]
Overall looks good, just couple of questions/comments.. Great job Kapil!

File: src/test/java/com/mathworks/ci/task/MatlabTestTaskTest.java (hunk 4 above)
Comment (mw-kapilg) [reply src/test/java/com/mathworks/ci/task/MatlabTestTaskTest.java:119]: - Here `Detailed` is just an input for testing which is later verified in line 141-142.
- The default level of verbosity would be same as the arguments' actual default (`Default` keyword is used for this purpose, similar to Jenkins).

File: src/main/resources/matlab-bamboo-plugin.properties (hunk 3 above)
Comment (mw-kapilg) [reply src/main/resources/matlab-bamboo-plugin.properties:38]: Sure, thanks!

**Review by ppandian28 — APPROVED** [review]

**Pull request merged**